jumeg/connectivity/con_viz.py

```diff
@@ -700,22 +708,51 @@ def _get_node_grouping(label_groups, orig_labels):
     return node_order, group_numbers
 
 
-def _get_group_node_angles_and_colors(label_groups, orig_labels, cortex_colors=None):
-    if cortex_colors is None:
-        cortex_colors = ['m', 'b', 'y', 'c', 'r', 'g',
-                         'g', 'r', 'c', 'y', 'b', 'm']
+def _get_group_node_angles(label_groups, orig_labels):
+    node_order, group_numbers = _get_node_grouping(label_groups, orig_labels)
+
+    # the respective no. of regions in each cortex
+    group_boundaries = np.cumsum([0] + group_numbers)[:-1]
+
+    node_angles = circular_layout(orig_labels, node_order, start_pos=90,
+                                  group_boundaries=group_boundaries)
+
+    return node_angles
+
+
+def _get_node_colors(label_groups, orig_labels, cortex_colors=None):
 
     node_order, group_numbers = _get_node_grouping(label_groups, orig_labels)
 
-    # assign a color and angle to each label based on the group
-    node_angles, node_colors = _get_node_angles_and_colors(group_numbers, cortex_colors,
-                                                           node_order, orig_labels)
+    if cortex_colors is None:
+        from matplotlib.pyplot import cm
+        n_colors = len(group_numbers)
+        cortex_colors = cm.gist_rainbow(np.linspace(0, 1, n_colors))
```

Comment:
Instead of hard-coding a colormap, you could allow user to specify a colormap.

```
def discrete_cmap(N, base_cmap=None):
    from matplotlib.colors import ListedColormap
    """Create an N-bin discrete colormap from the specified input map"""
    # Note that if base_cmap is a string or None, you can simply do
    #    return plt.cm.get_cmap(base_cmap, N)
    # The following works for string, None, or a colormap instance:
    base = plt.cm.get_cmap(base_cmap)
    color_list = base(np.linspace(0, 1, N))
    cmap_name = base.name + str(N)
    return color_list, ListedColormap(color_list, cmap_name, N)
```

Reply:
Then the user could also simply provide the cortex colors themselves. You need a colormap with lots of different colors so most colormaps won't work anyways. Furthermore, we already have a cmap argument for the color of the arrows.

Reply:
Agreed.